### PR TITLE
Ignore twilio error codes correctly

### PR DESCRIPTION
**Pull request by DeeTheDev**
Our [fix for twilio verification update error](https://github.com/lithictech/suma/pull/714) regressed. This is because we checked for an HTTP 404 error but failed to check for twilio API specific error code. Fix and test.

### CODE REVIEW

**Comment by rgalanakis** [comment]
What do you mean "regressed"?

**Comment by rgalanakis** [comment]
Also please link to the Sentry issue.

**Comment by DeeTheDev** [comment]
@rgalanakis The [issue log](https://lithic-technology.sentry.io/issues/5880814468/?alert_rule_id=12147465&alert_type=issue&notification_uuid=aab09819-b1c1-4b35-8d36-dd8d1472c069&project=6724127&referrer=slack) regressed in Sentry errors.

File: lib/suma/twilio.rb [hunk 1]
```diff
@@ -41,8 +41,12 @@ def self.update_verification(ve_id, kw)
       update(**kw)
     return response
   rescue Twilio::REST::RestError => e
-    # ignore 404s, it means twilio has approved, expired or invalidated the code already
+    # ignores 20404s, it means twilio has approved, expired or invalidated the code already
     # https://www.twilio.com/docs/verify/api/verification-check#check-a-verification
-    raise(e) unless e.code === 404
```
Comment (rgalanakis) [inline lib/suma/twilio.rb:46]: As a note- Twilio automatically set the `code` to `404`(http status) if there's no `code` in the body. That is why this test worked previously, even though `body.code` wasn't passed by the test.

**Pull request merged**